cli/util/cancel.go

```diff
@@ -0,0 +1,153 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/cli/flags"
+	"github.com/nspcc-dev/neo-go/cli/input"
+	"github.com/nspcc-dev/neo-go/cli/options"
+	"github.com/nspcc-dev/neo-go/cli/paramcontext"
+	cliwallet "github.com/nspcc-dev/neo-go/cli/wallet"
+	"github.com/nspcc-dev/neo-go/pkg/core"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/address"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/urfave/cli"
+)
+
+func cancelTx(ctx *cli.Context) error {
+	args := ctx.Args()
+	if len(args) == 0 {
+		return cli.NewExitError("missing input file", 1)
+	} else if len(args) > 2 {
+		return cli.NewExitError("only one input file is accepted", 1)
+	}
+	var (
+		tx  *transaction.Transaction
+		err error
+	)
+
+	h, err := util.Uint256DecodeStringLE(strings.TrimPrefix(args[0], "0x"))
+	if err != nil {
+		pc, err := paramcontext.Read(args[0])
+		if err != nil {
+			return fmt.Errorf("invalid tx hash or path to parameter context: %w", err)
+		}
+		var ok bool
+		tx, ok = pc.Verifiable.(*transaction.Transaction)
+		if !ok {
+			return errors.New("failed to retrieve transaction from parameter context: verifiable item is not a transaction")
+		}
+	} else {
+		bc := ctx.App.Metadata["chain"].(*core.Blockchain)
+		tx, _, err = bc.GetTransaction(h)
+		if err == nil {
+			return fmt.Errorf("too late cancel transaction it is in chain: %w", err)
+		}
+	}
+
+	gctx, cancel := options.GetTimeoutContext(ctx)
+	defer cancel()
+
+	c, err := options.GetRPCClient(gctx, ctx)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create RPC client: %w", err), 1)
+	}
+
+	acc, w, err := getAccFromContext(ctx)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	defer w.Close()
+
+	a, err := actor.NewSimple(c, acc)
+	script := byte(opcode.NOP)
+
+	attr := append(tx.Attributes, transaction.Attribute{Type: transaction.ConflictsT})
+
+	_, _, err = a.SendTunedRun([]byte{script}, attr, func(r *result.Invoke, t *transaction.Transaction) error {
+		err := actor.DefaultCheckerModifier(r, t)
+		if err != nil {
+			return err
+		}
+		t.NetworkFee = int64(math.Max(float64(tx.NetworkFee), float64(tx.SystemFee))) + 1
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getAccFromContext(ctx *cli.Context) (*wallet.Account, *wallet.Wallet, error) {
```

Comment:
Try to avoid code duplication. This method already exists in other package. Create a separate commit where this method becomes exported, put this commit before the `cli: cancel transaction command ` commit. And after that in the current commit use this exported method.

Reply:
You can use interactive git rebase to make `getAccFromContext` exported in a separate commit. Use `git rebase -i HEAD~2` and then mark the previous commit with `e`.
See the https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---interactive.